upload/views/commits.py

```diff
@@ -26,7 +26,14 @@
 log = logging.getLogger(__name__)
 
 
-class CommitViews(ListCreateAPIView, GetterMixin):
+class CommitLogicMixin:
+    def create_commit(self, serializer, repository):
```

Comment:
I've implemented the logic using functions instead of class method via mixins. Only challenge I had with doing this was the `get_token_for_analytics` function, which uses `self.request`. To avoid passing `self.request` through 4-5 functions, I passed to `get_token_for_analytics` first to get a `token`, and only pass this `token` to `create_upload` instead of passing the entire `self.request` to `create_upload`.